connection.go

```diff
@@ -317,9 +328,9 @@ func (s *Connection) Serve(newHandler StreamHandler) {
 			partition = partitionRoundRobin
 			partitionRoundRobin = (partitionRoundRobin + 1) % FRAME_WORKERS
 		case *spdy.GoAwayFrame:
-			priority = 0
-			partition = partitionRoundRobin
-			partitionRoundRobin = (partitionRoundRobin + 1) % FRAME_WORKERS
+			// handle to go-away frame synchronously here, then exit the loop
+			s.handleGoAwayFrame(frame)
```

Comment:
Yeah, it got put in a queue, so it was possible for other frames to be enqueued after a goaway